src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxUiaTextProvider.cs

```diff
@@ -0,0 +1,421 @@
+                clientLocation.Y = Math.Max(clientLocation.Y, clientRectangle.top);
+                clientLocation.Y = Math.Min(clientLocation.Y, clientRectangle.bottom);
+
+                // Get the character at those client coordinates.
+                int start = GetCharIndexFromPosition(clientLocation);
+
+                return new UiaTextRange(new InternalAccessibleObject(_owningComboBox.ChildEditAccessibleObject), this, start, start);
+            }
+
+            public override UiaCore.ITextRangeProvider DocumentRange => new UiaTextRange(new InternalAccessibleObject(_owningComboBox.ChildEditAccessibleObject), this, 0, TextLength);
+
+            public override UiaCore.SupportedTextSelection SupportedTextSelection => UiaCore.SupportedTextSelection.Single;
+
+            public override UiaCore.ITextRangeProvider? GetCaretRange(out BOOL isActive)
+            {
+                isActive = BOOL.FALSE;
+
+                if (!_owningComboBox.IsHandleCreated)
+                {
+                    return null;
+                }
+
+                var hasKeyboardFocus = _owningComboBox.ChildEditAccessibleObject.GetPropertyValue(UiaCore.UIA.HasKeyboardFocusPropertyId);
+                if (hasKeyboardFocus is bool && (bool)hasKeyboardFocus)
+                {
+                    isActive = BOOL.TRUE;
+                }
+
+                var internalAccessibleObject = new InternalAccessibleObject(_owningComboBox.ChildEditAccessibleObject);
+
+                return new UiaTextRange(internalAccessibleObject, this, _owningComboBox.SelectionStart, _owningComboBox.SelectionStart);
+            }
+
+            public override Point PointToScreen(Point pt) => _owningComboBox.PointToScreen(pt);
+
+            /// <summary>
+            ///  Exposes a text range that contains the text that is the target of the annotation associated with the specified annotation element.
+            /// </summary>
+            /// <param name="annotationElement">
+            ///  The provider for an element that implements the IAnnotationProvider interface.
+            ///  The annotation element is a sibling of the element that implements the <see cref="UiaCore.ITextProvider2"/> interface for the document.
+            /// </param>
+            /// <returns>
+            ///  A text range that contains the annotation target text.
+            /// </returns>
+            public override UiaCore.ITextRangeProvider RangeFromAnnotation(UiaCore.IRawElementProviderSimple annotationElement)
+            {
+                var internalAccessibleObject = new InternalAccessibleObject(_owningComboBox.ChildEditAccessibleObject);
+
+                return new UiaTextRange(internalAccessibleObject, this, 0, 0);
+            }
+
+            public override Rectangle BoundingRectangle
+                => _owningComboBox.IsHandleCreated
+                    ? (Rectangle)GetFormattingRectangle()
```

Comment:
We can make like that: 
```suggestion
                    ? GetFormattingRectangle()
```